src/main/java/fi/aalto/cs/apluscourses/intellij/model/task/IntelliJComponentPresenterBase.java

```diff
@@ -28,8 +33,16 @@ public void highlight() {
 
   @RequiresEdt
   private void highlightInternal() {
+    if (overlayPane != null) {
+      overlayPane.remove();
+    }
+
     GenericHighlighter highlighter = getHighlighter();
     if (highlighter == null) {
+      if (tryToShow()) {
+        highlightInternal();
```

Comment:
Possible livelock if `getHiglighter` always returns null and `tryToShow` true (well, not that much "livelock" as it ends up in a stack overflow quite soon). Can't the logic in `tryToShow` be handled by highlighter itself?

Reply:
the logic can't be in the highlighter, because the GenericHighlighter doesn'r have enough info about the component

Reply:
You fixed the livelock / stack overflow issue, anyway. :+1: